lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else if (callback) {
+    throw new TypeError('callback must be a function');
```

Comment:
We need that `if` guard if and only if `callback` might be optional and therefore `undefined`. I didn't look at the code to see if that is actually possible.